lucene/translate.py

```diff
@@ -154,36 +178,43 @@ def postprocess_translated_objc(path):
             if any(fnmatch.fnmatch(full_path_java, ptn) for ptn in excluded):
                 continue
 
-            full_path_m = full_path_java.replace(
-                src, dst).replace(".java", ".m")
+            full_path_m = full_path_java.replace(src, dst).replace(".java", ".m")  # nopep8
             if os.path.exists(full_path_m):
                 if os.path.getmtime(full_path_m) >= os.path.getmtime(full_path_java):  # nopep8
                     continue
+
             to_compile.append(full_path_java)
             to_postprocess.append(full_path_m)
 
-    print('%s => %s files to compile' % (src, len(to_compile)))
+    print('\nTranslating: %s with %d java files to compile.\n' % (src, len(to_compile)))  # nopep8
 
     if to_compile:
-        print('Compiling %d java files' % len(to_compile))
         args = [
-            'j2objc',
-            # '-use-arc',
-            '-classpath', classpaths,
-            '--segmented-headers',
-            '-sourcepath', src,
+            j2objc,
             '-d', dst,
+            '-classpath', ':'.join(classpaths),
+            '--swift-friendly',
+            '--nullability',
+            '--doc-comments',
+            '--no-extract-unsequenced'
```

Comment:
Let's remove this flag. Adding that flag causes J2ObjC to produce code like this (I'm using C instead of Objective-C, but you get the point):

```c
void f(int x, int y, int z);

int a = 0;
f(a++, a++, a++); // is this f(0, 1, 2) or f(2, 1, 0) ?
```

This is an infamous [undefined behavior](http://stackoverflow.com/questions/18729323/unsequenced-modification-and-access-to-parameter) in C.

We currently don't treat warnings as errors in our Xcode projects across the repos, but this is a Clang warning. If you build `ObjCLucene.xcodeproj` you'll see "Multiple unsequenced modifications to x" warnings here and there. Without the flag, J2ObjC will use temp variables to enforce the sequence point semantics of Java:

```c
jint a = 0;
jint unseq$1 = a++;
jint unseq$2 = a++;
```

Then use `unseq$1`, `unseq$2`, and finally `a`.

Could you re-translate the code without that flag and make a PR to ObjCLucene (and the related projects) again? Lucene is a big enough project that this kind of undefined behavior in C could cause very subtle errors that are very tricky to debug down the road, since you still get a working executable.